src/Confluent.Kafka/Producer.cs

```diff
@@ -425,42 +640,123 @@ public void Dispose()
 
+        /// <summary>
+        ///     Get information pertaining to all groups in the Kafka
+        ///     cluster.
+        ///
+        ///     [UNSTABLE-API] - The API associated with this functionality is subject to change.
+        /// </summary>
         public List<GroupInfo> ListGroups()
             => kafkaHandle.ListGroups(-1);
 
-
+        /// <summary>
+        ///     Get information pertaining to a particular group in the
+        ///     Kafka cluster.
+        ///
+        ///     [UNSTABLE-API] - The API associated with this functionality is subject to change.
+        /// </summary>
+        /// <param name="group">
+        ///     The group of interest.
+        /// </param>
+        /// <param name="timeout">
+        ///     The maximum period of time the call should block.
+        /// </param>
+        /// <returns>
+        ///     Returns information pertaining to the specified group
+        ///     or null if this group does not exist.
+        /// </returns>
         public GroupInfo ListGroup(string group, TimeSpan timeout)
             => kafkaHandle.ListGroup(group, timeout.TotalMillisecondsAsInt());
 
+        /// <summary>
+        ///     Get information pertaining to a particular group in the
+        ///     Kafka cluster.
+        ///
+        ///     [UNSTABLE-API] - The API associated with this functionality is subject to change.
+        /// </summary>
+        /// <param name="group">
+        ///     The group of interest.
+        /// </param>
+        /// <returns>
+        ///     Returns information pertaining to the specified group
+        ///     or null if this group does not exist.
+        /// </returns>
         public GroupInfo ListGroup(string group)
             => kafkaHandle.ListGroup(group, -1);
 
 
+        /// <summary>
+        ///     Query the Kafka cluster for low (oldest/beginning) and high (newest/end)
+        ///     offsets for the specified topic/partition.
+        ///
+        ///     [UNSTABLE-API] - The API associated with this functionality is subject to change.
+        /// </summary>
+        /// <param name="topicPartition">
+        ///     The topic/partition of interest.
+        /// </param>
+        /// <param name="timeout">
+        ///     The maximum period of time the call should block.
+        /// </param>
+        /// <returns>
+        ///     The requested WatermarkOffsets.
+        /// </returns>
         public WatermarkOffsets QueryWatermarkOffsets(TopicPartition topicPartition, TimeSpan timeout)
             => kafkaHandle.QueryWatermarkOffsets(topicPartition.Topic, topicPartition.Partition, timeout.TotalMillisecondsAsInt());
 
+        /// <summary>
+        ///     Query the Kafka cluster for low (oldest/beginning) and high (newest/end)
+        ///     offsets for the specified topic/partition.
+        ///
+        ///     [UNSTABLE-API] - The API associated with this functionality is subject to change.
```

Comment:
ok. this is actually noted in the timeout parameter. but it's probably worth having here (ditto other places)